pkg/storegateway/bucket.go

```diff
@@ -117,6 +117,9 @@ type BucketStore struct {
 	// Query gate which limits the maximum amount of concurrent queries.
 	queryGate gate.Gate
 
+	// Gate used to limit concurrency on loading index-headers across all tenants.
+	readerGate gate.Gate
```

Comment:
maybe this should be something like `indexHeaderGate`, `lazyLoadingGate`, or `headerLoadingGate`. I feel like `readerGate` can be confusing.